server/mscalendar/availability.go

```diff
@@ -41,16 +43,17 @@ func (m *mscalendar) Sync(mattermostUserID string) (string, error) {
 func (m *mscalendar) SyncAll() (string, error) {
 	err := m.Filter(withSuperuserClient)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("not able to filter the super user client. %w", err)
```

Comment:
I think this is a little cleaner. Same with other references of `fmt.Errorf`

```suggestion
		return "", errors.Wrap(err, "not able to filter the super user client")
```